lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -823,7 +823,8 @@ void swift::serialization::diagnoseSerializedASTLoadFailure(
         llvm::find_if(loadedModuleFile->getDependencies(),
                       [](const ModuleFile::Dependency &next) {
                         return next.isLoaded() &&
-                               !next.Import.second->hasResolvedImports();
+                               ! (next.Import.hasValue() && 
+                               		next.Import.getValue().importedModule->hasResolvedImports());
```

Comment:
You don't need a `hasValue()` check here, because the code earlier was already relying on the pointer being non-null (also `isLoaded` checks for `hasValue()`). You can simplify it to (using the `operator->` on Optional):

```
                        return next.isLoaded() &&
                               !(next.Import->importedModule->hasResolvedImports());
```

Reply:
Oh, I see! I missed that because I guess typing was my best subject in school. I'll change it.

Reply:
Resolved in 1d625d8